tests/python/unittest/test_arith_rewrite_simplify.py

```diff
@@ -293,6 +294,9 @@ def test_sub_index_simplify():
 
     # mul co-efficient foldng
     ck.verify(x - x, 0)
+    ck.verify(a - a, 0)
+    ck.verify(a - b, a - b)
+    ck.verify(a - b, c - c)
```

Comment:
Why could we verify `a-b == c-c == 0`? This doesn't make sense to me.

Reply:
In this line, `a - b` will not be simplified, it is still `a - b`, so `structural_equal((a - b), (c-c))` is True.  Just test the functionality of `structural_equal`

Reply:
Not simplifying `a - b` is correct, but I think `a - b` is not structural equal to `c - c` as `a` and `b` refer to two different variable but `c - c` are the same variable.

Reply:
Remove this line now. Semantically, `a - b` is not equal to `c - c`, but `structural_equal((a - b), (c-c))` works because for AnyNode it use the `SEqualReduce` function which only checks the node's dtype. In fact, a function contains AnyNode(s) in its `ret_type` after InferType will have new AnyNode(s), the two func then differs though they pass the tests.

I think a long term approach should be map `Any` like `Var` when checking the structures. So just keep it for now?